crates/bevy_reflect/src/impls/rect.rs

```diff
@@ -1,9 +1,18 @@
 use crate as bevy_reflect;
 use crate::prelude::ReflectDefault;
 use crate::{ReflectDeserialize, ReflectSerialize};
-use bevy_math::{Rect, Vec2};
+use bevy_math::{IRect, IVec2, Rect, URect, UVec2, Vec2};
 use bevy_reflect_derive::impl_reflect_struct;
 
+impl_reflect_struct!(
+    #[reflect(Debug, PartialEq, Serialize, Deserialize, Default)]
```

Comment:
Both of these types should also add `Hash` to this list, since they implement `Hash` (unlike `Rect`).

Reply:
Should `Hash` be put behind a `#[cfg_attr(not(target_arch = "spirv"), derive(Hash))]` attribute since `IVec2`/`UVec2` have that cfg?

The reason I didn't add it is because `IVec2`/`UVec2`'s `Debug` implementation is also `#[cfg_attr(not(target_arch = "spirv")]`, yet `Debug` is still derived directly on `IRect`/`URect`.

I also added `Eq` derives for convenience, but I couldn't add `Eq` to the reflect derive since the compiler complained about `ReflectEq`, which I couldn't find. Does there exist an `Eq` reflect trait?

Reply:
I see. Yeah I would try to match the same flags that the source code defines.

And no, there is no `ReflectEq`. `Eq` is not treated specially by the macro either (unlike `PartialEq`, `Debug`, and `Hash`).

Reply:
My guess is that bevy would just fail to compile for the spirv target architecture, since the Hash and Debug derive macro requirements wouldn't be satisfied. But bevy would probably already fail to compile on this arch for many other reasons, so I don't see this as a problem.